### ISSUE TITLE
Bug(perf): CN node panic when run nexmark q5-rewrite

### ISSUE THREAD

**Issue by huangjw806** [risingwavelabs/risingwave#9522]
Daily performance test pipeline failed.

CN node panic when run nexmark q5-rewrite query：

```
Execution URL: https://buildkite.com/risingwave-test/nexmark-benchmark/builds/931

Grafana: https://g-2927a1b4d9.grafana-workspace.us-east-1.amazonaws.com/d/EpkBw5W4k/risingwave-test-dashboard?orgId=1&from=1682619249000&to=1682619372000&var-namespace=nexmark-bs-3-4-5-5r-daily-20230427

Log: https://ap-southeast-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/$252Frisingwave$252Ftest-useast1-eks-a$252Fnexmark-bs-3-4-5-5r-daily-20230427
```

Key panic log:
```
"log": "2023-04-27T18:14:40.394994739Z stderr F thread 'risingwave-streaming-actor' panicked at 'called `Option::unwrap()` on a `None` value', /risingwave/src/stream/src/executor/top_n/top_n_cache.rs:513:54"
```

**Comment by lmatz** [comment]
probably fixed by https://github.com/risingwavelabs/risingwave/pull/9501? @soundOfDestiny but didn't merge into main yesterday

**Comment by soundOfDestiny** [comment]
> probably fixed by #9501? @soundOfDestiny but didn't merge into main yesterday

Yes, it has been fixed by #9501
